dan_layer/engine/tests/hello_world/src/lib.rs

```diff
@@ -20,35 +20,14 @@
 //  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 //  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-// TODO: we should only use stdlib if the template dev needs to include it e.g. use core::mem when stdlib is not
-// available
+use tari_template_macros::template;
 
-use common::{generate_abi, generate_main, TemplateImpl};
-use tari_template_abi::{encode_with_len, FunctionDef, Type};
+template! {
```

Comment:
The scoping is why we'll need to put something like  

```rust
#[tari::contract]
mod hello_world {
   pub struct HelloWorld { etc.}
}
```

If we want to avoid wrapping in `template! {...}`, unless there's another way to get the scoping (I can't think of any) - the main reason is that some editors don't do code any completion when wrapped in a "inline looking" macro like this.

Reply:
This is fine IMO

It also allows some flexibility, because you can also write your contract in `my_contract.rs`, and then in `lib.rs` have

```rust
#[tari::contract]
mod my_contract;
```